src/components/Menu_Management/Zmones_Tab/PeopleListComponents/Head.jsx

```diff
@@ -56,24 +56,23 @@ class Head extends Component {
     let newUsers = this.props.sortedUsers;
     const typedName = this.nameInput.current.value.toLowerCase();
     const typedTopic = this.topicInput.current.value.toLowerCase();
-    this.props.storeFilter('name', typedName);
-    this.props.storeFilter('surname', typedTopic);
+    this.props.storeFilter('firstName', typedName);
+    this.props.storeFilter('lastName', typedTopic);
     if (typedName.length > 0) {
-      newUsers = newUsers.filter(i => i.name.toLowerCase().match(typedName));
+      newUsers = newUsers.filter(i => i.firstName.toLowerCase().match(typedName));
     }
     if (typedTopic.length > 0) {
-      newUsers = newUsers.filter(i => i.surname.toLowerCase().match(typedTopic));
+      newUsers = newUsers.filter(i => i.lastName.toLowerCase().match(typedTopic));
     }
     this.props.filterData(newUsers);
   }
 
   onSort(e, orderBy) {
     if (e !== null) e.preventDefault();
     //sets parameters
-    if (orderBy == 'name') var order = !this.props.nameDirection;
-    if (orderBy == 'surname') var order = !this.props.surnameDirection;
-    if (orderBy == 'hours') var order = !this.props.hoursDirection;
-    if (orderBy == 'activity') var order = !this.props.activityDirection;
+    if (orderBy == 'firstName') var order = !this.props.nameDirection;
+    if (orderBy == 'lastName') var order = !this.props.surnameDirection;
+    if (orderBy == 'available') var order = !this.props.activityDirection;
     if (orderBy == '') return;
```

Comment:
Try using `===` instead of `==` - double-equals tend to be unpredictable sometimes

Reply:
Sure, will change than in a minute